File: Lyft Backend Software Engineering/tires/tires_type/octoprime_tires.py
from tires.tires import Tires

# OctoprimeTires class inherits from Tires and represents Octoprime tires.
class OctoprimeTires(Tires):
    # Constructor initializes OctoprimeTires with tire wear data and service threshold.
    def __init__(self, tire_wear):
        self.tire_wear = tire_wear
        self.tire_need_to_service = 3.0

    # needs_service method checks if the sum of tire wear values meets the service threshold.
    def needs_service(self):
        return sum(self.tire_wear) >= self.tire_need_to_service

File: Softstraw Back-End Engineer/tires/tires_type/octoprime_tires.java
package tires.tires_type;

import tires.Tires;

// OctoprimeTires class inherits from Tires and represents Octoprime tires.
public class OctoprimeTires extends Tires {

    // Field to store tire wear data and service threshold.
    private double[] tireWear;
    private final double tireNeedToService = 3.0;

    // Constructor initializes OctoprimeTires with tire wear data.
    public OctoprimeTires(double[] tireWear) {
        this.tireWear = tireWear;
    }

    // needsService method checks if the sum of tire wear values meets the service threshold.
    @Override
    public boolean needsService() {
        return sum(tireWear) >= tireNeedToService;
    }

    // Helper method to calculate the sum of an array of doubles.
    private double sum(double[] array) {
        double total = 0.0;
        for (double value : array) {
            total += value;
        }
        return total;
    }
}
